### PR TITLE
Add user documentation

### LINKED ISSUE #14: Add proper documentation
Add user-documentation.

### PR DESCRIPTION
**Pull request by neithernut**
WIP. Resolves #14, #37.

### CODE REVIEW

File: doc/Motivation.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+case conceivable. You end up writing lots of repetitive code and potentially
+buggy or outright broken code, which, as explained above, should be avoided
+
+Some frameworks generate code for (de)serialization and require you to feed it
+every single piece of data manually. This may cause you to depend on a specific
+version of a specific code generator and, again, requires the user to write a
+lot of repetitive code.
+
+Then there are frameworks that either require you to add a super class to the
+thing you want to serialize or make use of code-annotations and compiler-plugins
```
Comment (TheNeikos) [inline doc/Motivation.md]: What do you consider the problem with code-annotations?

Comment (neithernut) [inline doc/Motivation.md]: You need to modify the source of the class. And if they are not of a very general nature, they require either code-generation or compiler-plugins. Will update soonish.

**Comment by neithernut** [comment]
I'll squash many of those commits at some point.

**Comment by neithernut** [comment]
@TheNeikos, @matthiasbeyer: I'd appreciate your input on this. I'd especially be interested in whether the stuff I wrote is easy to understand.
I'll also add some pandoc-setup. The chapters of the documentation will be concatenated in the following order:
- Introduction
- Motaivation
- Properties
- Factories
- AccessorBundle
- Utilities
- Licensing

I'm not quite sure where to put the "Usage" chapter, yet. Maybe right after "Motivation"?

**Comment by neithernut** [comment]
Argh... I clicked the wrong button again.

**Comment by matthiasbeyer** [comment]
> I'm not quite sure where to put the "Usage" chapter, yet. Maybe right after "Motivation"?

Definitively. Maybe after "Properties", depending on what's exactly in this chapter (I will read now).

File: doc/Motivation.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+object, either with a freshly generated one or, if the object is loaded from
+some source, with an existing UUID. You don't want to set it after the
+construction. You may not even be trivially able to.
+
+
+Conclusion
+----------
+
+What I always wanted was an unitrusive library which does not impose
+dependencies on the data type, does is not entirely based on preprocessors, code
```
Comment (matthiasbeyer) [inline doc/Motivation.md]: > does is not entirely based on preprocessors

this is not intended, is it?

File: doc/Properties.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    struct attribute;
+
+In addition to the standard property parameters `KeyType` and `Key`, the
+template also takes the `Type` of the attribute as a parameter. `Type` may also
+be `const` to denote read-only attributes. In this case, `<attribute>::is_const`
+is true, otherwise it's false. The original `Type` is exported as the member
+type `<attribute>::type`, with eventual `const` specifiers removed.
+
+It is not required that an attribute actually exists in a specific C++ type. An
+attribute can aw well be a deduced value. However, the attribute's type has to
```
Comment (matthiasbeyer) [inline doc/Properties.md]: typo: `aw well`

File: doc/Properties.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+be `const` in this case.
+
+The following overloads to the `accessor()` factory provided by attributes
+exist. However, some are available for const/non-const attributes only:
+ * `accessor<typename ObjType>(
+        Value(ObjType::* getter)(),
+        void(ObjType::* setter)(Value)
+    )`
```
Comment (matthiasbeyer) [inline doc/Properties.md]: Make sure this renders correctly in pandoc if you want to generate pdf out of this. I'm not sure whether this is valid markdown-codeblock syntax.

File: doc/Utilities.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+header `<cmoh/utils.hpp>`, in the `cmoh::util` namespace.
+
+
+`cmoh::optional`
+----------------
+
+Additionally, CMOH ships with an partial implementation of the `std::optional`
+template, but in the `cmoh` namespace. However, for post-C++14 builds the
+headers `<optional>` and `<experimental/optional>` are searched for. If found,
+`cmoh::optional` will be an alias of either `cmoh::optional` or
```
Comment (matthiasbeyer) [inline doc/Utilities.md:15]: > cmoh::optional`will be an alias of either`cmoh::optional`

not sure whether this is intended. It sounds like `alias foo=foo` ... which doesn't make that much sense, does it?

Comment (neithernut) [reply doc/Utilities.md:15]: [Type alias](http://en.cppreference.com/w/cpp/language/type_alias) is a term which actually describes a very similar expression. It's `using foo=bar`.

File: doc/Properties.md (hunk 4 above)
Comment (neithernut) [reply doc/Properties.md]: Thanks for the hint

**Pull request merged**